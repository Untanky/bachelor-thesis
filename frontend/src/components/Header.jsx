import React from 'react';

import pjson from '../../package.json';

import './Header.scss';

const Header = () => {
  return (
    <header className="header">
      <h1>Web&shy;schnitt&shy;stellen - Ver&shy;gleich von ver&shy;schie&shy;denen Pro&shy;gram&shy;mier&shy;spra&shy;chen und Frame&shy;works zur Er&shy;stel&shy;lung von REST-Schnitts&shy;tellen</h1>
      <div className="additional-info">
        <b>Bachelorarbeit von <a href="https://lukasgrimm.me">Lukas Grimm</a></b>
        <i>an der <a href="https://www.htw-berlin.de">Hochschule f√ºr Technik und Wirtschaft Berlin</a></i>
        <span>Version: {pjson.version}</span>
      </div>
    </header>
  )
};

export default Header;
