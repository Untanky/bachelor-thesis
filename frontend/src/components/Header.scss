@import "../theme";

.header {
  max-width: 1200px;
  padding: 1em 2em;
  border-bottom: 4px solid $htw-green;
  margin: 0 auto;

  h1 {
    font-size: 52px;
  }

  .additional-info {
    text-align: left;

    > * {
      display: block;
      padding: 0.25em 0;
    }
  }
}
