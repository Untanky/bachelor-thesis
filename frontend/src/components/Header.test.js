import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Header />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
