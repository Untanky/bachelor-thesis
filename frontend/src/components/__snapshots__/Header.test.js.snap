// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  class="header"
>
  <h1>
    Web­schnitt­stellen - Ver­gleich von ver­schie­denen Pro­gram­mier­spra­chen und Frame­works zur Er­stel­lung von REST-Schnitts­tellen
  </h1>
  <div
    class="additional-info"
  >
    <b>
      Bachelorarbeit von 
      <a
        href="https://lukasgrimm.me"
      >
        Lukas Grimm
      </a>
    </b>
    <i>
      an der 
      <a
        href="https://www.htw-berlin.de"
      >
        Hochschule für Technik und Wirtschaft Berlin
      </a>
    </i>
    <span>
      Version: 
      0.1.0
    </span>
  </div>
</header>
`;
